examples/gno.land/r/demo/profile/profile.gno

```diff
@@ -0,0 +1,312 @@
+		}
+	}
+
+	// Update fields and log changes
+	updateField(&existingProfile.Username, username, "Username", &existingProfile)
+	updateField(&existingProfile.AvatarURL, avatarurl, "AvatarURL", &existingProfile)
+	updateField(&existingProfile.Age, age, "Age", &existingProfile)
+	updateField(&existingProfile.Gender, gender, "Gender", &existingProfile)
+	updateField(&existingProfile.Website, website, "Website", &existingProfile)
+	updateField(&existingProfile.Country, country, "Country", &existingProfile)
+	updateField(&existingProfile.City, city, "City", &existingProfile)
+
+	// Update the profile in the tree
+	profiles.Set(caller.String(), existingProfile)
+
+	// Update the username index if the username is changed
+	if username != "" {
+		usernameIndex.Set(username, caller.String())
+	}
+}
+
+// Display complete Profile by Address
+func DisplayProfileByAddress(address std.Address) (UserProfile, bool) {
+	if value, exists := profiles.Get(address.String()); exists {
+		return value.(UserProfile), true
+	}
+	return UserProfile{}, false
+}
+
+// Get complete profile by username
+func DisplayProfileByUsername(username string) (UserProfile, bool) {
+	if addr, exists := usernameIndex.Get(username); exists {
+		return DisplayProfileByAddress(std.Address(addr.(string)))
+	}
+	return UserProfile{}, false
+}
+
+// Display username
+func DisplayUsername(address std.Address) string {
+	if value, exists := profiles.Get(address.String()); exists {
+		return value.(UserProfile).Username
+	}
+	return "Address not found"
+}
+
+// Display address from username
+func DisplayAddress(username string) string {
+	if addr, exists := usernameIndex.Get(username); exists {
+		return addr.(string)
+	}
+	return "Username not found"
+}
+
+// Display avatarurl from address or username
+func DisplayAvatarURL(address std.Address, username string) string {
+	if address != "" {
+		if value, exists := profiles.Get(address.String()); exists {
+			return value.(UserProfile).AvatarURL
+		}
+	} else if username != "" {
+		if addr, exists := usernameIndex.Get(username); exists {
+			if value, exists := profiles.Get(addr.(string)); exists {
+				return value.(UserProfile).AvatarURL
+			}
+		}
+	}
+	return "Address or Username not found"
+}
+
+// Display age from address or username
+func DisplayAge(address std.Address, username string) string {
+	if address != "" {
+		if value, exists := profiles.Get(address.String()); exists {
+			return value.(UserProfile).Age
+		}
+	} else if username != "" {
+		if addr, exists := usernameIndex.Get(username); exists {
+			if value, exists := profiles.Get(addr.(string)); exists {
+				return value.(UserProfile).Age
+			}
+		}
+	}
+	return "Address or Username not found"
+}
+
+// Display gender from address or username
+func DisplayGender(address std.Address, username string) string {
+	if address != "" {
+		if value, exists := profiles.Get(address.String()); exists {
+			return value.(UserProfile).Gender
+		}
+	} else if username != "" {
```

Comment:
You don't necessarily need an `else if`. Instead only using an `if` can solve your thing.